### PR TITLE
Fix issue with missing result for programming exercises in exam mode after refreshing the browser

### LINKED ISSUE #2779: "No graded results" for programming exercises after refreshing page in exams
### Describe the bug

When participating in an exam and refreshing the page after submitting a programming exercise it displays "no graded results" on the top of the page.

#### To Reproduce

1. Create an exam with a programming exercise
2. Participate as a student. Make some changes and submit. Your submission should be graded (also visible on the top). 
3. Refresh Artemis.
4. See, that "no graded results" is shown instead of former result.

#### Expected behavior

The former results should be displayed instead of "no graded results"

#### Screenshots

image

### Environment

<details><pre>
-OS: Windows
-tested on all widely used browsers
-Release: 4.9.3
</pre></details>

### PR DESCRIPTION
**Pull request by krusche**
Fixes #2779 which was introduced by unintended changes in #2738
Also improve REST URLs to follow the standard and to distinguish those better

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).

### Steps for Testing
1. Create an exam and change the exercise group order
2. Participate as student in an exam and double check that everything is still working (some REST Urls were changed)
3. Submit code for one programming exercise and wait until the results appears
4. Reload the page
5. Double check that the result is displayed correctly
6. Also try out a test run and make sure nothing is broken here

### CODE REVIEW

**Review by airelawaleria — COMMENTED** [review]
Tested on ts1 with multiple exams and test users, but the issue was the opposite for me now. 
After the first submit I always got "No graded result" and after refreshing, the tests were triggered for the fisrt time to run ("Building and testing") and then the actual result appeared. (Second reload still displayed the results correctly)

**Review by JohannesWeiss — COMMENTED** [review]
Tested with multiple exams and users on ts1 as well and observed the same behaviour as @airelawaleria. After reloading the results did show up as expected.
Test Runs seemed to work fine, no issues were apparent with those.

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java [hunk 1]
```diff
@@ -488,14 +489,7 @@ private void fetchParticipationsSubmissionsAndResultsForStudentExam(StudentExam
         // connect & filter the exercises and student participations including the latest submission and results where necessary, to make sure all relevant associations are
         // available
         for (Exercise exercise : studentExam.getExercises()) {
-            if (forConduction) {
```
Comment (derLalla) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:491]: `forConduction` needs to be removed from the function parameters, as it is not used anymore.

**Review by derLalla — CHANGES_REQUESTED** [review]
The code looks good. I have just found one unused code reference.

**Review by simonzilker — COMMENTED** [review]
Tested with @marvinchimeka . Noticed the same behavior as mentioned from @airelawaleria.

**Review by marvinchimeka — COMMENTED** [review]
Tested in testing session. Have nothing to add to @simonzilker statement.

**Review by philgabr — COMMENTED** [review]
Submission couldn't be graded for me (artemis_test_user_2, ts3):
image

**Review by sebastiantpeter — COMMENTED** [review]
For me the correct submission result was available on ts2.

**Review by kloessst — CHANGES_REQUESTED** [review]
Code looks fine and the result is correctly displayed after a browser reload now.

I also experienced the weird problems with receiving the new result over websocket, especially with the first submission after starting the exam. I don't see any changes regarding websockets on client or server side in this PR so this might be unrelated. I will still reject the PR for now until we get more tests and/or find the root cause of this problem.

**Review by philgabr — COMMENTED** [review]
I deployed build 6 on ts3 and although deployment was successful, it continously displays "Artemis is starting up" & a 502 is thrown 
image

**Review by airelawaleria — APPROVED** [review]
tested on ts1. worked as expected now

**Review by tiffca — COMMENTED** [review]
*edit* I think I had to refresh my browser, testing again
--
I couldn't participate in the exam at all. On TS1 after registering students and generating exam/prepare exercise start, the students receive this message when trying to participate "You are not registered for the exam . Please contact your instructor." Even though I can see the prepared exam for the students from the instructor side.

**Review by JohannesWeiss — APPROVED** [review]
Retested on ts1, worked as expected. The previous issue occurred only after logging out as creator of the exam and logging in as student/ test-user without a refresh of the page in between. After such a page reload, everything worked for me.

**Review by fde312 — APPROVED** [review]
Tested it and it works fine. Code looks good as well 👍

**Pull request merged**